content/documentation/pages/4-batch-developer-guides/2-batch/2-spring-batch.md

```diff
@@ -8,34 +8,310 @@ description: 'Create a Spring Batch Job'
+- Start/Stop/Restart
+- Retry/Skip
+- Resource management
+
+It also provides more advanced technical services and features that will enable extremely high-volume and high performance batch jobs through optimization and partitioning techniques.
 
-Create a single step batch job, file item reader, file item writer, no other dependencies outside of spring batch.
+For this guide we will focus on 5 Spring Batch components:
+
+![BILL_STATMENTS](images/spring-batch-reference-model.png)
+
+- Job: A `job` is an entity that encapsulates an entire batch process. A job is comprised of one or more `steps`.
+- Step: A `Step` is a domain object that encapsulates an independent, sequential phase of a batch job. Each `step` is comprised of a `ItemReader`, `ItemProcessor`, and a `ItemWriter`.
+- ItemReader: `ItemReader` is an abstraction that represents the retrieval of input for a Step, one item at a time.
+- ItemProcessor: `ItemProcessor` is an abstraction that represents the business processing of an item.
+- ItemWriter: `ItemWriter` is an abstraction that represents the output of a Step
+
+In the diagram above we see that each phase of the `JobExecution` is stored into a `JobRepository` (our MySql database). This means that each action performed by Spring Batch is recorded to a database for both logging purposes but also for restarting a job.
+
+NOTE: You can read more about this process [here](https://docs.spring.io/spring-batch/4.0.x/reference/html/domain.html#domainLanguageOfBatch).
+
+### Our Batch Job
+
+So for our application we will have a BillRun `Job` that will have one `Step` that will comprised of:
+
+- JsonItemReader: Is an `ItemReader` that will read a Json file containing the usage data.
+- BillProcessor: Is an `ItemProcessor` that will generate a price based on each row of data sent from the JsonItemReader.
+- JdbcBatchItemWriter: Is an `ItemWriter` that will write the priced Bill record to the `BILLING_STATEMENT` table.
 
 ### Initialzr
 
-Go to initializr - maybe have a curl command ready to go do we can just download a ready made .zip.
+1. Visit the [Spring Initialzr site](https://start.spring.io/).
+1. Select the latest release of spring boot.
+1. Create a new Maven project with a Group name of `io.spring` and an Artifact name of `billrun`.
+1. In the Dependencies text box, type `task` to select the Cloud Task dependency.
+1. In the Dependencies text box, type `jdbc` then select the JDBC dependency.
+1. In the Dependencies text box, type `h2` then select the H2 dependency.
+   1. We use H2 for unit testing.
+1. In the Dependencies text box, type `mysql` then select mysql dependency (or your favorite database).
+   1. We use MySql for the runtime database.
+1. In the Dependencies text box, type `Batch` then select Batch.
+1. Click the Generate Project button.
+1. Unzip the billrun.zip file and import the project into your favorite IDE.
+
+Another option instead of using the UI to initialize your project you can do the following:
```

Comment:
I like this way of displaying the direct download option rather than the way we have it in the Task doc...please fix to be consistent.